### PR TITLE
chore: Tweak debug symbols generation for releases

### PR DESCRIPTION
**Pull request by royjacobson**
A proposed fix for #1136.

@Pothulapati, what's the best way to test that this works?

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,8 +32,6 @@ build-debug:
 package:
 	cd build-opt; \
 	mv dragonfly $(RELEASE_NAME); \
-	tar cvfz $(RELEASE_NAME).unstripped.tar.gz $(RELEASE_NAME) ../LICENSE.md; \
-	strip $(RELEASE_NAME); \
```
Comment (Pothulapati) [inline Makefile:36]: If we don't strip this here? Do we need fall into the same problem of https://github.com/dragonflydb/dragonfly/issues/999? 🤔

**Comment by Pothulapati** [comment]
@royjacobson Because we have two different Dockerfiles for development and prod (Yep, Not ideal), The ideal way would be to test them both.
- First, You would test if adding the flags actually added the debug symbols into the binary and the resultant docker image. For this you can run `DOCKER_BUILDKIT=1 docker build -f ./tools/docker/Dockerfile.ubuntu-dev . -t ghcr.io/dragonfly/df-debug` and test out the `ghcr.io/dragonfly/df-debug` image.
- The release pipeline uses `make package` which has its own processing. For this, You can run a weekly release workflow on your branch by going to https://github.com/dragonflydb/dragonfly/actions/workflows/docker-weekly.yml, and running on your branch. This would publish an image into our packages repo for you to use. Because this is internal we should be fine!

File: Makefile (hunk 1 above)
Comment (royjacobson) [reply Makefile:36]: The build flags change is doing two things - it reduces the 'level' of the debug symbols from 2 to 1, and it tells gcc to compress them. So we get a binary with debug symbols and a reasonable size of 16MB which we can use unstripped.

Comment (Pothulapati) [reply Makefile:36]: Ah Nice!

**Comment by royjacobson** [comment]
Looks great! The size of the docker image I created locally was 95MB. The CI action failed but I think it was on some external gchr error not related. @Pothulapati

**Pull request merged**